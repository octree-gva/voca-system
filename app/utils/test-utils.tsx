import React, {ReactElement, FC} from 'react';
import {ThemeProvider} from '@mui/material/styles';
import theme from '../theme';
import {Session} from 'next-auth';
import {SessionProvider} from 'next-auth/react';
import {render, RenderOptions} from '@testing-library/react';

export type TestWrapperProps = React.PropsWithChildren<{
  session: Session | null;
}>;
const TestWrapper: FC<TestWrapperProps> = ({children, session}) => {
  return (
    <SessionProvider session={session}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </SessionProvider>
  );
};
const UnAuthenticatedWrapper: FC = props => (
  <TestWrapper {...props} session={null} />
);
const AuthenticatedWrapper: FC = props => (
  <TestWrapper
    {...props}
    session={{user: {username: 'john-doe'}, expires: `${+new Date() + 30000}`}}
  />
);

const customRender = (
  ui: ReactElement,
  options?: {isPublic?: boolean} & Omit<RenderOptions, 'wrapper'>
) =>
  render(ui, {
    wrapper: !!options?.isPublic
      ? UnAuthenticatedWrapper
      : AuthenticatedWrapper,
    ...options,
  });

export * from '@testing-library/react';
// Override our render with the snapshot-friendly render.
export {customRender as render};
