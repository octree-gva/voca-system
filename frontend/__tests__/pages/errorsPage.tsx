import {render, screen} from 'test-utils';
import NextRouter from 'next/router';
import Errors from '../../pages/errors';

describe('Errors page renders', () => {
  let fragment: DocumentFragment | null;

  describe('status=404', () => {
    beforeEach(() => {
      jest.spyOn(NextRouter, 'useRouter').mockReturnValue({
        route: '/errors/404',
        query: {errorCode: '404'},
      } as any);
      const {asFragment} = render(<Errors />);
      fragment = asFragment();
    });
    it('matches snapshot', () => {
      expect(fragment).toMatchSnapshot();
    });
    it('show i18n 404.title key', () => {
      const errorTitle = screen.getByText(/errors.404.title/i);
      expect(errorTitle).toBeDefined();
    });
    it('show i18n 404.desc key', () => {
      const errorDesc = screen.getByText(/errors.404.desc/i);
      expect(errorDesc).toBeDefined();
    });
  });

  describe('status=502', () => {
    beforeEach(() => {
      jest.spyOn(NextRouter, 'useRouter').mockReturnValue({
        route: '/errors/502',
        query: {errorCode: '502'},
      } as any);
      const {asFragment} = render(<Errors />);
      fragment = asFragment();
    });
    it('matches snapshot', () => {
      expect(fragment).toMatchSnapshot();
    });
    it('show i18n 502.title key', () => {
      const errorTitle = screen.getByText(/errors.502.title/i);
      expect(errorTitle).toBeDefined();
    });
    it('show i18n 502.desc key', () => {
      const errorDesc = screen.getByText(/errors.502.desc/i);
      expect(errorDesc).toBeDefined();
    });
  });
});
