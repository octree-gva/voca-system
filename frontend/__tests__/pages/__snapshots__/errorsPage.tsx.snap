// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors page renders status=404 matches snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-width: 320px;
  max-width: 100vw;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  margin: 0;
}

.emotion-1 {
  padding: 8px;
  margin-top: 16px;
  margin-bottom: 32px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  margin-top: 32px;
  margin-bottom: 32px;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-5 {
  margin: 10px 0;
  font-size: 26px;
  font-weight: bold;
  font-family: "Inter",sans-serif;
  line-height: 1.2;
  margin-top: 32px;
  margin-bottom: 32px;
}

.emotion-6 {
  margin: 0;
  font-size: .875rem;
  font-weight: bold;
  font-family: "Inter",sans-serif;
  line-height: 1.6;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Inter",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-height: auto;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 20px;
  padding: 0 16px;
  min-width: 40px;
  width: auto;
  height: 40px;
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
  color: rgba(0, 0, 0, 0.87);
  background-color: #e0e0e0;
  color: #fff;
  background-color: #253550;
  position: fixed;
  right: 16px;
  bottom: 16px;
  text-transform: initial;
  z-index: 1301;
}

.emotion-7::-moz-focus-inner {
  border-style: none;
}

.emotion-7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-7:active {
  box-shadow: 0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12);
}

.emotion-7:hover {
  background-color: #f5f5f5;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (hover: none) {
  .emotion-7:hover {
    background-color: #e0e0e0;
  }
}

.emotion-7.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-7.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-7:hover {
  background-color: #192538;
}

@media (hover: none) {
  .emotion-7:hover {
    background-color: #253550;
  }
}

<div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <div
        class="MuiBox-root emotion-2"
      >
        <a
          class="MuiButtonBase-root emotion-3"
          href="/"
          tabindex="0"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27240%27%20height=%2760%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="metas.title"
              data-nimg="intrinsic"
              decoding="async"
              src="/logo.png"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </span>
          <span
            class="MuiTouchRipple-root emotion-4"
          />
        </a>
      </div>
      <h2
        class="MuiTypography-root MuiTypography-h2 emotion-5"
      >
        errors.404.title
      </h2>
      <h6
        class="MuiTypography-root MuiTypography-h6 emotion-6"
      >
        errors.404.desc
      </h6>
      <a
        aria-label="support"
        class="MuiButtonBase-root MuiFab-root MuiFab-extended MuiFab-sizeMedium MuiFab-primary emotion-7"
        href="mailto:support@octree.ch"
        tabindex="0"
      >
        support
        <span
          class="MuiTouchRipple-root emotion-4"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Errors page renders status=502 matches snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-width: 320px;
  max-width: 100vw;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0;
  margin: 0;
}

.emotion-1 {
  padding: 8px;
  margin-top: 16px;
  margin-bottom: 32px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  margin-top: 32px;
  margin-bottom: 32px;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-5 {
  margin: 10px 0;
  font-size: 26px;
  font-weight: bold;
  font-family: "Inter",sans-serif;
  line-height: 1.2;
  margin-top: 32px;
  margin-bottom: 32px;
}

.emotion-6 {
  margin: 0;
  font-size: .875rem;
  font-weight: bold;
  font-family: "Inter",sans-serif;
  line-height: 1.6;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Inter",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-height: auto;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 20px;
  padding: 0 16px;
  min-width: 40px;
  width: auto;
  height: 40px;
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
  color: rgba(0, 0, 0, 0.87);
  background-color: #e0e0e0;
  color: #fff;
  background-color: #253550;
  position: fixed;
  right: 16px;
  bottom: 16px;
  text-transform: initial;
  z-index: 1301;
}

.emotion-7::-moz-focus-inner {
  border-style: none;
}

.emotion-7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-7:active {
  box-shadow: 0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12);
}

.emotion-7:hover {
  background-color: #f5f5f5;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (hover: none) {
  .emotion-7:hover {
    background-color: #e0e0e0;
  }
}

.emotion-7.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-7.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-7:hover {
  background-color: #192538;
}

@media (hover: none) {
  .emotion-7:hover {
    background-color: #253550;
  }
}

<div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <div
        class="MuiBox-root emotion-2"
      >
        <a
          class="MuiButtonBase-root emotion-3"
          href="/"
          tabindex="0"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27240%27%20height=%2760%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="metas.title"
              data-nimg="intrinsic"
              decoding="async"
              src="/logo.png"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </span>
          <span
            class="MuiTouchRipple-root emotion-4"
          />
        </a>
      </div>
      <h2
        class="MuiTypography-root MuiTypography-h2 emotion-5"
      >
        errors.502.title
      </h2>
      <h6
        class="MuiTypography-root MuiTypography-h6 emotion-6"
      >
        errors.502.desc
      </h6>
      <a
        aria-label="support"
        class="MuiButtonBase-root MuiFab-root MuiFab-extended MuiFab-sizeMedium MuiFab-primary emotion-7"
        href="mailto:support@octree.ch"
        tabindex="0"
      >
        support
        <span
          class="MuiTouchRipple-root emotion-4"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
